aws-android-sdk-appsync-compiler/src/main/kotlin/com/apollographql/apollo/compiler/Util.kt

```diff
@@ -24,7 +24,7 @@ fun String.escapeJavaReservedWord() = if (JAVA_RESERVED_WORDS.contains(this)) "$
 fun TypeName.overrideTypeName(typeNameOverrideMap: Map<String, String>): TypeName {
   if (this is ParameterizedTypeName) {
     val typeArguments = typeArguments.map { it.overrideTypeName(typeNameOverrideMap) }.toTypedArray()
-    return ParameterizedTypeName.get(rawType, *typeArguments)
+    return ParameterizedTypeName.get(rawType.overrideTypeName(typeNameOverrideMap) as ClassName, *typeArguments)
```

Comment:
nit: did you mean to do `as?` for null safe cast?

Reply:
If null, what should be the TypeName in case of `as?`?

Reply:
sure!  I'm certain that it's going to be `ClassName` because `overrideTypeName` returns the same type as it gets, but the compiler doesn't know that.